### ISSUE TITLE
[Logic nodes] Math assign node

### ISSUE THREAD

**Issue by ItsCubeTime** [armory3d/armory#2224]
The purposeof these nodes would be to shorten node graphs slightly by removing the need of having a separate math node.

Im sure a lot of people might think it seem excessive. Personally I think its worth a node of its own considering how often you might just want to increment or decrement a variable by 1, a node like this could help a lot.

Most programming languages has custom assignment operators like +=, *=, -= (etc) which lets you modify and assign a variable in one line without having to write out the variable name twice - the purpose of this node would be pretty much the same.

UE4s popular Blueprint system has something similar (but less generic): https://answers.unrealengine.com/storage/temp/56085-notify.png

In practicse, the node would have the same inputs/outputs as the existing math node, except that it would also assign the output value to the variable that is fed into the first input. Alternatively this could also be an tickbox "assign to first output" on the already existing math node.

**Comment by MoritzBrueckner** [comment]
> Alternatively this could also be an tickbox "assign to first output" on the already existing math node.

That sounds good :)

**Comment by ItsCubeTime** [comment]
> > Alternatively this could also be an tickbox "assign to first output" on the already existing math node.
> 
> That sounds good :)

Yes, I added that part because I knew you were gonna suggest it elsewise 😂
